staging/src/k8s.io/dynamic-resource-allocation/resourceslice/noderesources.go

```diff
@@ -0,0 +1,392 @@
+		// Dump entire resource information.
+		loggerV.Info("Syncing existing driver resource slices with driver resources", "slices", klog.KObjSlice(slices), "resources", resources)
+	} else {
+		logger.V(5).Info("Syncing existing driver resource slices with driver resources", "slices", klog.KObjSlice(slices), "numResources", len(resources.NodeResources))
+	}
+
+	// Retrieve node object to get UID?
+	// The result gets cached and is expected to not change while
+	// the controller runs.
+	if c.owner.UID == "" && c.owner.APIVersion == "v1" && c.owner.Kind == "Node" {
+		node, err := c.kubeClient.CoreV1().Nodes().Get(ctx, c.owner.Name, metav1.GetOptions{})
+		if err != nil {
+			return fmt.Errorf("retrieve node %q: %v", c.owner.Name, err)
+		}
+		// There is only one worker, so no locking needed.
+		c.owner.UID = node.UID
+	}
+
+	// Update stale slices before removing what's left.
+	//
+	// We don't really know which of these slices might have
+	// been used for "the" driver resource because they don't
+	// have a unique ID. In practice, a driver is most likely
+	// to just give us one ResourceModel, in which case
+	// this isn't a problem at all. If we have more than one,
+	// then at least conceptually it currently doesn't matter
+	// where we publish it.
+	//
+	// The long-term goal is to move the handling of
+	// ResourceSlice objects into the driver, with kubelet
+	// just acting as a REST proxy. The advantage of that will
+	// be that kubelet won't need to support the same
+	// resource API version as the driver and the control plane.
+	// With that approach, the driver will be able to match
+	// up objects more intelligently.
+	numObsoleteSlices := len(obsoleteSlices)
+	for index, resource := range resources.NodeResources {
+		if storedResourceIndices.Has(index) {
+			// No need to do anything, it is already stored exactly
+			// like this in an existing slice.
+			continue
+		}
+
+		if numObsoleteSlices > 0 {
+			// Update one existing slice.
+			slice := obsoleteSlices[numObsoleteSlices-1]
+			numObsoleteSlices--
+			slice = slice.DeepCopy()
+			slice.ResourceModel = *resource
+			logger.V(5).Info("Reusing existing resource slice", "slice", klog.KObj(slice))
+			if _, err := c.kubeClient.ResourceV1alpha2().ResourceSlices().Update(ctx, slice, metav1.UpdateOptions{}); err != nil {
+				return fmt.Errorf("update resource slice: %w", err)
+			}
+			continue
+		}
+
+		// Create a new slice.
+		slice := &resourceapi.ResourceSlice{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: c.owner.Name + "-" + c.driverName + "-",
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						APIVersion: c.owner.APIVersion,
+						Kind:       c.owner.Kind,
+						Name:       c.owner.Name,
+						UID:        c.owner.UID,
+						Controller: ptr.To(true),
+					},
+				},
+			},
+			DriverName:    c.driverName,
+			ResourceModel: *resource,
+		}
+		if c.owner.APIVersion == "v1" && c.owner.Kind == "Node" {
+			slice.NodeName = c.owner.Name
+		}
```

Comment:
What about the pool info?

Reply:
Edit: This PR sits below the changes to the API, so the Pool info isn't here yet....